go/vt/sidecardb/sidecardb.go

```diff
@@ -140,25 +140,48 @@ func printCallerDetails() {
 type vtSchemaInit struct {
 	ctx            context.Context
 	exec           Exec
+	sroHook        SetSuperReadOnlyHook
+	rsroHook       ReSetSuperReadOnlyHook
 	existingTables map[string]bool
 	dbCreated      bool // the first upgrade/create query will be prefixed by a _vt creation script
 }
 
 // Exec is the prototype of a callback that has to be passed to Init() to execute the specified query in the database
 type Exec func(ctx context.Context, query string, maxRows int, wantFields bool, useVT bool) (*sqltypes.Result, error)
+type SetSuperReadOnlyHook func(ctx context.Context) (needsReset bool, err error)
+type ReSetSuperReadOnlyHook func(ctx context.Context) (err error)
 
 // GetDDLCount metric returns the count of sidecardb ddls that have been run as part of this vttablet's init process.
 func GetDDLCount() int64 {
 	return ddlCount.Get()
 }
 
 // Init creates or upgrades the _vt sidecar database based on declarative schema for all _vt tables
-func Init(ctx context.Context, exec Exec) error {
+func Init(ctx context.Context, exec Exec, sroHook SetSuperReadOnlyHook, rsroHook ReSetSuperReadOnlyHook) error {
```

Comment:
we seem to be sending nil in all invocations of this call that I looked at.

Reply:
This doesn't seem like something that would ever be variable. Curious why we would use hooks/callbacks? To me, it feels more like a boolean:
  - Init w/o SRO handling
  - Init with SRO handling
And any time we change a variable, we reset it in a defer.

Reply:
I wanted to remove that. This code was initially written by me when we don't had primary check in schema/engine.go. I think we don't need this anymore, so I pushed the change initially with nil. Now since we don't see any failure I will remove it in my next commit.